Async Communication (#55)
loadsmart/culture: practices/communication.md

# Communication

## Async Communication

At Loadsmart, the engineering team works 100% remotely. We believe async communication is key for our day-to-day work.

### Business Time Visibility

> There are people working from different time zones and we need to be aware of and respect it. For example, if we don't know your timezone or your business time, we don't know when you would be able to attend a meeting or to answer a message. Two tips here:

Reviewer: ```suggestion
People are working from different time zones, and we need to be aware of and respect it. For example, if we don't know your timezone or your business time, we don't know when you would be able to attend a meeting or answer a message. Two tips here:
```

> You can set your business time using Google Calendar. With that, everybody will be able to see it when scheduling a meeting with you.

Reviewer: ```suggestion
You can set your business time using Google Calendar. With that, everybody will see it when scheduling a meeting with you.
```

Add this information to your Slack profile. It will be available with a single click on your Slack picture. We recommend this format `<job title> @ <squad/team/department> - Working hours: X AM - Y PM (<timezone>)`

### No Hello

When sending a message, like in Slack, we usually try to be polite by not jumping right into the request. This is not a good practice and costs time. The [No Hello](https://nohello.net) website presents a better way to do it. Please take a look.
<div align="center">
  <img alt="No Hello" src="./no-hello.png" width="600" />
  <br/>
  <i>Image by <a href="https://nohello.net">No Hello</a></i>
</div>

### Slack Notifications

> It is common and useful to have Slack installed on our phones. With this we got the notifications as well. They are very useful when we are working, but they can be very unhealthy. Receiving notifications when you are out of your business time, will prevent you from disconnecting and resting. Within some time it will cause a lot of stress. That's why we recommend setting Slack to send mobile notifications during your business hours or for really urgent channels only.

Reviewer: ```suggestion
It is common and useful to have Slack installed on our phones. With this, we got the notifications as well. They are very useful when we are working, but they can be very unhealthy. Receiving notifications when you are out of your business time will prevent you from disconnecting and resting. Within some time, it will cause a lot of stress. That's why we recommend setting Slack to send mobile notifications during your business hours or for really urgent channels only.
```

### Slack Status

Slack is your front door. People will look for you there when they need to ask or tell something. It's important to make clear if you are available or not. For that, we can use the Slack Status. For example: Out of Office, In a Meeting, Internet Connection Unstable. Setting the integration between Google Calendar and Slack will update your status automatically.

### Collaborative Docs

We have many meetings to discuss ideas or proposals. Instead of scheduling a meeting whenever a new project or problem arises, try writing about it in a Google Doc, so the team can use the comments features to have an async discussion. This way, we give people time to understand and evaluate the subject, and they can make a decision without trying to fit everyone in a call at the same time. This feature is also available in other tools like Notion.

### How to Solve a Fire

> It takes time to figure out how to solve a fire and there is always a lesson learning. This learning should be shared so that we save time in the future. We recommend, after solving a fire, to write a how to fix it. The next firefighters will thank you for that.

Reviewer: ```suggestion
It takes time to figure out how to solve a fire, and there is always a lesson learning. This learning should be shared so that we save time in the future. After solving a fire, we recommend writing a how to fix it. The next firefighters will thank you for that.
```

### Meetings Outcome

> With the company getting bigger and bigger, it's more likely that people won't be able to attend all the meetings they are invited to. Because of that, it's recommended to either record and/or take notes and share it after the meeting. Those who missed it can get up to date afterward. Even for recorded meetings, that is a good idea to summarize it into notes so people can save time avoiding watching the whole video. Also we recommend using the Meetings Notes feature available on Google Calendar.

Reviewer: ```suggestion
With the company getting bigger and bigger, it's more likely that people won't be able to attend all the meetings they are invited to. Because of that, it's recommended to either record and/or take notes and share them after the meeting. Those who missed it can get up to date afterward. Even for recorded meetings, that is a good idea to summarize it into notes so people can save time avoiding watching the whole video. Also, we recommend using the Meetings Notes feature available on Google Calendar.
```

### Emails

Urgent communications should not happen via email. With that being said, we do not need to read an email as soon as it gets in our inbox. The recommendation is to set in the calendar/agenda one or two times per day to read emails. Make it a routine.

### Lock your Calendar

> Organize you calendar by locking some time for lunch or other regular breaks you have. Also lock it when you will be OOO. This will prevent you from receiving meeting invites when you cannot attend it. Company wide meetings may still happen in these times. It is up to you to organize your time according to your priorities, both personal and at the company.

Reviewer: ```suggestion
Organize your calendar by locking some time for lunch or other regular breaks you have. Also, lock it when you will be OOO. This will prevent you from receiving meeting invites when you cannot attend them. Company-wide meetings may still happen in these times. It is up to you to organize your time according to your priorities, both personal and at the company.
```

At Loadsmart, the engineering team works 100% remotely. We believe async communication is key for our day-to-day work.

### Business Time Visibility

People are working from different time zones, and we need to be aware of and respect it. For example, if we don't know your timezone or your business time, we don't know when you would be able to attend a meeting or answer a message. Two tips here:
You can set your business time using Google Calendar. With that, everybody will see it when scheduling a meeting with you.
Add this information to your Slack profile. It will be available with a single click on your Slack picture. We recommend this format `<job title> @ <squad/team/department> - Working hours: X AM - Y PM (<timezone>)`

### No Hello

When sending a message, like in Slack, we usually try to be polite by not jumping right into the request. This is not a good practice and costs time. The [No Hello](https://nohello.net) website presents a better way to do it. Please take a look.
<div align="center">
  <img alt="No Hello" src="./no-hello.png" width="600" />
  <br/>
  <i>Image by <a href="https://nohello.net">No Hello</a></i>
</div>

### Slack Notifications

> It is common and useful to have Slack installed on our phones. With this, we got the notifications as well. They are very useful when we are working, but they can be very unhealthy. Receiving notifications when you are out of your business time will prevent you from disconnecting and resting. Within some time, it will cause a lot of stress. That's why we recommend setting Slack to send mobile notifications during your business hours or for really urgent channels only.

Reviewer 2: ```suggestion
It is common to have Slack installed on our phones. But that comes with notifications too. Notifications are useful while you are working, but they are certainly unhealthy when you get bombarded by them outside of working hours. It prevents you from disconnecting and resting. Within time, it may increase your level of stress. That's why we recommend configuring Slack to send mobile notifications during your business hours only or for messages sent in urgent channels.
```

### Slack Status

> Slack is your front door. People will look for you there when they need to ask or tell something. It's important to make clear if you are available or not. For that, we can use the Slack Status. For example: Out of Office, In a Meeting, Internet Connection Unstable. Setting the integration between Google Calendar and Slack will update your status automatically.

Reviewer 2: ```suggestion
Slack is our main communication tool. It's through there that people will reach out to you when they need to ask or tell you something. It's crucial to make it clear when you are available and when you are not. For that, we use Slack Status. Some examples are: 

- ⛔ `Out of Office`
- 📆 `In a Meeting`
- 🌩️ `Unstable Internet Connection`
- etc

Tip: configuring the integration between Google Calendar and Slack will also update your status automatically whenever you are in a meeting.
```

### Slack Threads

Threads can be used to organize discussions in Slack. Multiple subjects can flood channels with messages. With threads, reply messages can be tied to their original message keeping readers notified with every new message and without distracting others in the channel that might not have an interest in that particular subject. 

### Collaborative Docs

We have many meetings to discuss ideas or proposals. Instead of scheduling a meeting whenever a new project or problem arises, try writing about it in a Google Doc, so the team can use the comments features to have an async discussion. This way, we give people time to understand and evaluate the subject, and they can make a decision without trying to fit everyone in a call at the same time. This feature is also available in other tools like Notion.

> ### How to Solve a Fire

Reviewer 2: ```suggestion
### Debugging urgent issues a.k.a. "How to Solve a Fire"
```

> It takes time to figure out how to solve a fire, and there is always a lesson learning. This learning should be shared so that we save time in the future. After solving a fire, we recommend writing a how to fix it. The next firefighters will thank you for that.

Reviewer 2: ```suggestion
There are fires of different sizes. Yet there's always lessons to be learned. This learning should always be shared with the team so that we save time in the future. Whenever you solve one of those difficult fires, try to cover that case with a regression test. We also recommend doing a write up on how to fix it, if applicable. The next firefighters will thank you for that.
```

### Meetings Outcome

> With the company getting bigger and bigger, it's more likely that people won't be able to attend all the meetings they are invited to. Because of that, it's recommended to either record and/or take notes and share them after the meeting. Those who missed it can get up to date afterward. Even for recorded meetings, that is a good idea to summarize it into notes so people can save time avoiding watching the whole video. Also, we recommend using the Meetings Notes feature available on Google Calendar.

Reviewer 2: ```suggestion
With the company getting bigger and bigger, it's more likely that people won't be able to attend all the meetings they are invited to. Because of that, it's recommended to either record and/or take notes and to share them after the meeting. Those who could not make it can get up to date afterward. Even for recorded meetings, it is a good idea to summarize it into notes so that people can save time by not having to watch the entire recording.

And when creating a meeting, please make use of the Meetings Notes section of the invite to clearly describe what's it for.
```

### Emails

> Urgent communications should not happen via email. With that being said, we do not need to read an email as soon as it gets in our inbox. The recommendation is to set in the calendar/agenda one or two times per day to read emails. Make it a routine.

Reviewer 2: ```suggestion
Urgent communications should not happen via email. With that being said, people should also not expect that others read their emails as soon as they hit their inboxes. The recommendation is to set a time in your calendar - once or twice a day - to read emails. Make it a routine.
```

> ### Lock your Calendar

Reviewer 2: ```suggestion
### Block time in your Calendar
```

> Organize your calendar by locking some time for lunch or other regular breaks you have. Also, lock it when you will be OOO. This will prevent you from receiving meeting invites when you cannot attend them. Company-wide meetings may still happen in these times. It is up to you to organize your time according to your priorities, both personal and at the company.

Reviewer 2: ```suggestion
Organize your calendar by blocking some time for lunch and other regular breaks you may need. Also, block it when you are Out-of-office. This will prevent you from receiving invitations for meetings that you cannot attend. 

On the other hand, company-wide meetings (Monthly, Town Hall, etc) may still be scheduled and conflict with your blocked slots. In those cases, do your best to attend those as important communication happens there.

At the end, it is up to you to organize your time according to your priorities, both personal and at the company.
```
